core/src/main/clojure/xtdb/metadata.clj

```diff
@@ -351,6 +354,27 @@
         (when (.getObject bloom-rdr bloom-vec-idx)
           (bloom/bloom->bitmap bloom-rdr bloom-vec-idx)))))
 
+  (temporalBounds[_ page-idx]
+    ;; TODO when badly distributed there are pages with no data
```

Comment:
At some point I was running into the case where the tries were pointing to empty pages with badly distributed data. I don't know if we should consider this a bug or if such cases should filtered elsewhere in the trie merging logic.

Reply:
yes, ideally - when the page is created. there's some logic here already, IIRC, but clearly it isn't catching a certain case...

Reply:
See https://github.com/xtdb/xtdb/issues/3580